server/bin/test/test_pbench_server_put_shim_001.py

```diff
@@ -0,0 +1,150 @@
+        return lines[-1].replace("\n", "")
+
+    def test_get_receive_dir(self):
+        """verify the pbench-receive-dir-prefix directory"""
+        res = get_receive_dir(self.config, self.logger)
+        expected_path = "/srv/pbench/pbench-move-results-receive/fs-version-002"
+
+        assert str(res).endswith(expected_path)
+
+    def test_get_receive_dir_failed(self, invalid_config):
+        """verify the pbench-receive-dir-prefix directory.
+
+        Note: Checking logs to look for any failure"""
+
+        res = get_receive_dir(invalid_config, self.logger)
+        expected_error_msg = "Failed: No value for config option pbench-receive-dir-prefix in section pbench-server"
+        log = self.log_output()
+
+        assert res is None
+        assert log.endswith(expected_error_msg)
+
+    def test_get_receive_dir_value_failed(self, invalid_value_config):
+        res = get_receive_dir(invalid_value_config, self.logger)
+        prefx = invalid_value_config.get("pbench-server", "pbench-receive-dir-prefix")
+        expected_error_msg = (
+            f"Failed: {prefx}-002 does not exist, or is not a directory"
+        )
+        log = self.log_output()
+
+        assert res is None
+        assert log.endswith(expected_error_msg)
+
+    def test_process_tb_file_not_found_error(self):
+        """checks normal processing when tar ball is not present"""
+
+        receive_dir = "server/bin/test/fixtures/upload/bad_tarball"
+        expected_result = Results(nstatus="", ntotal=1, ntbs=0)
+        expected_error_msg = (
+            f"No such file or directory: '{receive_dir}/bad/bad_tb.tar.xz'"
+        )
+        res = process_tb(self.config, self.logger, receive_dir)
+        log = self.log_output()
+
+        assert res == expected_result
+        assert log.endswith(expected_error_msg)
+
+    def test_process_tb_connection_error(self, monkeypatch):
+        """checks normal processing when Connection Error is faced"""
+
+        def mock_copy_result_tb(self, token):
+            raise RuntimeError(f"Cannot connect to '{self.upload_url}'")
+
+        receive_dir = "server/bin/test/fixtures/upload/tarball"
+        expected_result = Results(nstatus="", ntotal=1, ntbs=0)
+        expected_error_msg = (
+            "Cannot connect to 'https://pbench.example.com/v2/1/upload/log.tar.xz'"
+        )
+        monkeypatch.setattr(CopyResultTb, "copy_result_tb", mock_copy_result_tb)
+        res = process_tb(self.config, self.logger, receive_dir)
+        log = str(self.log_output())
+
+        assert res == expected_result
+        assert log.endswith(expected_error_msg)
+
+    def test_bad_tb(self):
+        """checks Results(namedtuple) return from process_tb module when any
+            kind of error is raised."""
+
+        receive_dir = "server/bin/test/fixtures/upload/bad_tarball"
+        expected_error_msg = Results(nstatus="", ntotal=1, ntbs=0)
+        res = process_tb(self.config, self.logger, receive_dir)
```

Comment:
I think that this scenario is an (inferior) duplicate of `test_process_tb_file_not_found_error()`, so you should just remove this scenario (unless you can make it into something new and better).